boa/vyper/abi.py

```diff
@@ -0,0 +1,32 @@
+from vyper.semantics.types import DArrayT, EventT, VyperType
+from vyper.semantics.types.utils import type_from_abi
+
+
+class ABIEvent(EventT):
+    @classmethod
+    def from_abi(cls, abi: dict) -> "ABIEvent":
+        """
+        Generate an `Event` object from an ABI interface.
+
+        Arguments
+        ---------
+        abi : dict
+            An object from a JSON ABI interface, representing an event.
+
+        Returns
+        -------
+        Event object.
+        """
+        return cls(
+            name=abi["name"],
+            arguments={items["name"]: parse_abi_type(items) for items in abi["inputs"]},
+            indexed=[i["indexed"] for i in abi["inputs"]],
+        )
+
+
+def parse_abi_type(abi: dict) -> VyperType:
```

Comment:
i think maybe a more robust method will be to go directly from the ABI dict/json format to an ABI (string) signature. i used the vyper parser as a shortcut to be compatible with the `VyperFunction` machinery but if we need to rewrite the `prepare_calldata/__call__` machinery think we can just bypass it (parsing to vyper types) at this point.

Reply:
i think `eth-abi` or `eth-stdlib` (both included as dependencies already) should have some utilities for this. also may want to wrap the dependency as is done in `boa/util/abi.py`

Reply:
What do you think about  [this](https://github.com/vyperlang/titanoboa/blob/71058ad9c430322b9de7e838f82d1dd81d3a20dc/boa/util/abi.py#L29)?